pkg/drivers/qemu/qemu.go

```diff
@@ -451,13 +461,20 @@ func (d *Driver) Start() error {
 			d.diskPath())
 	}
 
-	// If socket network, start with socket_vmnet.
+	// for socket_vmnet network, start with socket_vmnet
 	startProgram := d.Program
 	if d.Network == "socket_vmnet" {
 		startProgram = d.SocketVMNetClientPath
 		startCmd = append([]string{d.SocketVMNetPath, d.Program}, startCmd...)
 	}
 
+	// vmnet network requires elevated privileges
+	if strings.HasPrefix(d.Network, "vmnet-") {
+		//TODO: handle windows
+		startProgram = "sudo"
```

Comment:
maybe change this to "sudo -e"

Reply:
@medyagh sorry for not coming back sooner to this, but i didn't have a mac at hand to test earlier - now i do :)

i was finally able to replicate the issue you described above (not sure what's the difference between this and prev mac i tried on and did not have that problem before)

i tried with `-E` (i guess you meant with the capital `e`) - ie, the `preserve environment` flag, but the issue remained

in the latest commit, i opted to retain the user's PATH explicitly, and it worked - can you please pull & try and let me know if that worked for you as well

before:
```
% minikube start --driver qemu --network vmnet-shared
😄  minikube v1.30.1 on Darwin 13.5 (arm64)
✨  Using the qemu2 driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...\
sudo password:-
sudo password:
sudo password:  OUTPUT:
ERROR: Password:
sudo: qemu-system-aarch64: command not found


🔥  Deleting "minikube" in qemu2 ...
🤦  StartHost failed, but will try again: creating host: create: creating: Password:
sudo: qemu-system-aarch64: command not found: exit status 1
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...|
sudo password:/
sudo password:
sudo password:  OUTPUT:
ERROR: Password:
sudo: qemu-system-aarch64: command not found


😿  Failed to start qemu2 VM. Running "minikube delete" may fix it: creating host: create: creating: Password:
sudo: qemu-system-aarch64: command not found: exit status 1

❌  Exiting due to GUEST_PROVISION: error provisioning guest: Failed to start host: creating host: create: creating: Password:
sudo: qemu-system-aarch64: command not found: exit status 1

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯
```

after:
```
% minikube start --driver qemu --network vmnet-shared
😄  minikube v1.30.1 on Darwin 13.5 (arm64)
✨  Using the qemu2 driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...\
sudo password:
❗  This VM is having trouble accessing https://registry.k8s.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  Preparing Kubernetes v1.26.3 on Docker 20.10.23 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```